p2pserver/net/netserver/netserver.go

```diff
@@ -259,15 +260,23 @@ func (this *NetServer) IsPeerEstablished(p *peer.Peer) bool {
 		return this.Np.NodeEstablished(p.GetID())
 	}
 	return false
-
 }
 
 //Connect used to connect net address under sync or cons mode
 func (this *NetServer) Connect(addr string, isConsensus bool) error {
+	this.Lock()
+	defer this.Unlock()
+	connCount := this.GetOutConnectingListLen()
+	if connCount > config.DefConfig.P2PNode.MaxConnOutBound {
```

Comment:
AFAIK, GetOutConnectingListLen returns the number of nodes in handshaking state, not the total number of connected nodes.

Reply:
If here used total number of connected nodes, attacker can also creates many connections which under "handshaking" status but not really connected before the connected nodes reached the upper limit,  socket handles will be used over limit first.

Reply:
`total number of connected nodes`, here I mean  `number of handshaking nodes + number of  handshaking completed nodes`.

Reply:
I've fixed it, thx~